testsuite/integration-arquillian/servers/app-server/jetty/common/pom.xml

```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0"?>
+<!--
+~ Copyright 2018 Red Hat, Inc. and/or its affiliates
+~ and other contributors as indicated by the @author tags.
+~
+<?xml version="1.0"?>
+~ http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing, software
+~ distributed under the License is distributed on an "AS IS" BASIS,
+~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~ See the License for the specific language governing permissions and
+~ limitations under the License.
+-->
+
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <parent>
+        <groupId>org.keycloak.testsuite</groupId>
+        <artifactId>integration-arquillian-servers-app-server-jetty</artifactId>
+        <version>4.8.3.Final-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>integration-arquillian-servers-app-server-jetty-common</artifactId>
+    <packaging>jar</packaging>
```

Comment:
Here's the interesting part for you guys.

As you can see the `common` module consists of some mandatory dependencies (like the `keycloak-saml-servlet-filter-adapter`) and a set of dependencies marked as `<optional>true</optional>`. Optional dependencies are not transitive. Therefore, it is up to `integration-arquillian-servers-app-server-jetty-9x` (and `integration-arquillian-servers-app-server-jetty-81`) modules to set the dependencies properly but the implementation is shared at the same time. If you want to see this in action, please have a look that I'm using a different set of dependencies for 8 and 9 around security as well as JSPs. 

Similar thing happens to the tests. As you can see, the `common` module contains basic "sanity" tests (just to check if app server implementation looks OK-ish), which are reused by specific modules.